Assets/BossRoom/Scripts/Client/UI/RoomNameBox.cs

```diff
@@ -4,10 +4,16 @@
 using UnityEngine.Assertions;
 using TMPro;
 using Unity.Netcode;
+using UnityEngine.UI;
 
 public class RoomNameBox : MonoBehaviour
 {
-    [SerializeField] TextMeshProUGUI m_RoomNameText;
+    [SerializeField]
+    TextMeshProUGUI m_RoomNameText;
+    [SerializeField]
+    Button m_CopyToClipboardButton;
+
+    string m_RoomName;
```

Comment:
is there a possibility that CopyToClipboard will be clicked before m_RoomNameText.text is initialized? I'm not a big fan of storing data in gui text, but in this case here, we're literally asking to copy the content of the text box, why not just use m_RoomNameText to store the room name? (I'm asking so we have one less state to maintain)

Reply:
I don't think there is, since m_RoomNameText.text is initialized in Awake(), and then is set in ConnectionFinished(), before the button is enabled.
As for m_RoomName, I put the code in a different variable to follow the structure we had with Photon for m_RoomNameText.text ($"Room Name: {roomName}") which I find is clearer for players.

Reply:
oh right I get what you mean. nvm then :)